airflow/www/api/experimental/endpoints.py

```diff
@@ -234,3 +235,20 @@ def delete_pool(name):
         return response
     else:
         return jsonify(pool.to_json())
+
+
+@csrf.exempt
+@api_experimental.route('/dags/<string:dag_id>/remove_dag', methods=['POST'])
+@requires_authentication
+def remove_dag(dag_id):
+    """
+    Remove all DB records related to the specified Dag.
+    """
+    try:
+        count = remove.remove_dag(dag_id)
+    except Exception as err:
+        _log.error(err)
+        response = jsonify(error="{}".format(err))
+        response.status_code = 404
```

Comment:
How do you know it is a 404? I could be a 500 error. Please make your exception handling less broad